src/plugins/discover/public/application/main/services/discover_saved_search_container.ts

```diff
@@ -0,0 +1,267 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import {
+  getEmptySavedSearch,
+  getSavedSearch,
+  SavedSearch,
+  saveSavedSearch,
+} from '@kbn/saved-search-plugin/public';
+import { BehaviorSubject } from 'rxjs';
+import type { DataView } from '@kbn/data-views-plugin/common';
+import { SavedObjectSaveOpts } from '@kbn/saved-objects-plugin/public';
+import { isEqual } from 'lodash';
+import { restoreStateFromSavedSearch } from '../../../services/saved_searches/restore_from_saved_search';
+import { updateSavedSearch } from '../utils/update_saved_search';
+import { addLog } from '../../../utils/add_log';
+import { handleSourceColumnState } from '../../../utils/state_helpers';
+import { DiscoverAppState } from './discover_app_state_container';
+import { DiscoverServices } from '../../../build_services';
+import { getStateDefaults } from '../utils/get_state_defaults';
+
+export interface UpdateParams {
+  nextDataView?: DataView | undefined;
+  nextState?: DiscoverAppState | undefined;
+  filterAndQuery?: boolean;
+}
+
+/**
+ * Container for the saved search state, allowing to load, update and persist the saved search
+ * Can also be used to track changes to the saved search
+ * It centralizes functionality that was spread across the Discover main codebase
+ */
+export interface DiscoverSavedSearchContainer {
```

Comment:
Dear reviewer, my I introduce the new (sub) state container to you ... most of the code in this PR is about using is ... but this is the central part